lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,448 @@
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+    ---
+module: luks_device
+
+short_description: Manage encrypted (LUKS) devices
+
+version_added: "2.8"
+
+description:
+    - "Module manages LUKS on given device. Supports creating, destroying,
+      opening and closing of LUKS container and adding or removing new keys."
+
+options:
+    device:
+        description:
+            - "Device to work with (e.g. C(/dev/sda1)). Needed in most cases.
+              Can be omitted only when I(open=false) and I(name) is provided."
+        type: str
+    state:
+        description:
+            - "Based on its value creates or destroys the LUKS container on
+              a given device. Container creation requires I(device) and I(key)
+              options to be provided. Removal requires I(device) or I(name)."
+        default: present
+        choices: ['absent', 'present']
+        type: str
+    open:
+        description:
+            - "Based on its value unlocks or locks an existing LUKS container
+              on a given I(device). Use the I(name) option to set the name of
+              the opened container. Otherwise the name will be generated
+              automatically and returned in the result. Unlocking the container 
+              requires I(key) to be set."
+        default: false
+        type: bool
+    name:
+        description:
+            - "Sets container name when I(open=true). Can be used
+              instead of I(device) when closing the container
+              (i.e. when I(open=false))."
+        type: str
+    key:
+        description:
+            - "Used to unlock the container and needed for most
+              of the operations. Parameter value is the path
+              to the keyfile with the passphrase.
+              BEWARE that working with keyfiles in plaintext is dangerous.
+              Make sure that they are protected."
+        type: path
+    new_key:
+        description:
+            - "Adds additional key to given container on I(device).
+              Needs I(key) option for authorization. LUKS container
+              supports up to 8 keys. Parameter value is the path
+              to the keyfile with the passphrase.
+              BEWARE that working with keyfiles in plaintext is dangerous.
+              Make sure that they are protected."
+        type: path
+    remove_key:
+        description:
+            - "Removes given key from the container on I(device).
+              Parameter value is the path to the keyfile with the passphrase.
+              BEWARE that it is possible to remove even the last key from the
+              container. Data in there will be irreversibly lost.
+              BEWARE that working with keyfiles in plaintext is dangerous.
+              Make sure that they are protected."
+        type: path
+
+requirements:
+    - "cryptsetup, wipefs, lsblk"
```

Comment:
It's probably better to put each requirement into its own item.

Reply:
Fixed.